modules/services/repositories/src/main/java/au/com/shiftyjelly/pocketcasts/repositories/playback/CacheWorker.kt

```diff
@@ -0,0 +1,106 @@
+package au.com.shiftyjelly.pocketcasts.repositories.playback
+
+import android.content.Context
+import android.net.Uri
+import androidx.annotation.OptIn
+import androidx.hilt.work.HiltWorker
+import androidx.media3.common.util.UnstableApi
+import androidx.media3.datasource.DataSpec
+import androidx.media3.datasource.cache.CacheDataSource
+import androidx.media3.datasource.cache.CacheWriter
+import androidx.work.Constraints
+import androidx.work.Data
+import androidx.work.NetworkType
+import androidx.work.OneTimeWorkRequest
+import androidx.work.WorkManager
+import androidx.work.Worker
+import androidx.work.WorkerParameters
+import dagger.assisted.Assisted
+import dagger.assisted.AssistedInject
+import timber.log.Timber
+
+@OptIn(UnstableApi::class)
+@HiltWorker
+class CacheWorker @AssistedInject constructor(
+    @Assisted private val context: Context,
+    @Assisted params: WorkerParameters,
+    private val exoPlayerHelper: ExoPlayerHelper,
+) : Worker(context, params) {
+    private var cacheWriter: CacheWriter? = null
+
+    override fun doWork(): Result {
+        try {
+            val url = inputData.getString(URL_KEY)
+            val episodeUuid = inputData.getString(EPISODE_UUID_KEY)
+            val uri = Uri.parse(url)
+
+            val dataSourceFactory = exoPlayerHelper.getDataSourceFactory()
+            var dataSpec = DataSpec(uri)
+            if (episodeUuid != null) {
+                dataSpec = dataSpec.buildUpon().setKey(episodeUuid).build()
+            }
+
+            val cacheDataSourceFactory = CacheDataSource.Factory()
+                .setUpstreamDataSourceFactory(dataSourceFactory)
+                .setFlags(CacheDataSource.FLAG_IGNORE_CACHE_ON_ERROR)
+            exoPlayerHelper.getSimpleCache()?.let {
+                cacheDataSourceFactory.setCache(it)
+            }
+
+            cacheWriter = CacheWriter(
+                cacheDataSourceFactory.createDataSource(),
+                dataSpec,
+                null,
+            ) { requestLength: Long, bytesCached: Long, _: Long ->
+                if (bytesCached == requestLength) {
+                    Timber.d("$TAG: Caching complete for episode id: $episodeUuid")
+                }
+            }
+            cacheWriter?.cache()
```

Comment:
Logging can be moved after call to `cache()` because `cache()` is blocking. It simplifies the code and this way it doesn't have to rely on `requestLength` which might be unknown in some cases, which would make `bytesCached == requestLength` always `false`.

```suggestion
            cacheWriter = CacheWriter(
                cacheDataSourceFactory.createDataSource(),
                dataSpec,
                null,
                null,
            )
            cacheWriter?.cache()
            Timber.tag(TAG).d("Caching complete for episode id: $episodeUuid")
```